modules/core/src/main/java/com/google/refine/importers/TabularImportingParserBase.java

```diff
@@ -226,30 +215,8 @@ static public void readTable(
                     }
                 }
             }
-            if (!storeBlankColumns) {// if user don't choose storeBlankColumns, delete all empty columns.
-                deleteEmptyColumns(columnsHasData, project);
-            }
         } catch (IOException e) {
             exceptions.add(e);
-        } catch (ModelException e) {
-            exceptions.add(e);
-        }
-    }
-
-    /**
-     * If "storeBlankColumns" == false, delete blank columns.
-     *
-     * @param columnsHasData
-     *            Record if there is data in each column( false:null;true:has data)
-     */
-    static public void deleteEmptyColumns(List<Boolean> columnsHasData, Project project) throws ModelException {
-        project.columnModel.update(); // make sure all our cell indexes are up to date
-        for (int c = 0; c < columnsHasData.size(); c++) {
-            if (!columnsHasData.get(c)) {
-                // remove column from columns
-                project.columnModel.removeColumnByCellIndex(c);
-            }
         }
-        project.columnModel.update();
     }
```

Comment:
```suggestion
     /**
     * If "storeBlankColumns" == false, delete blank columns.
     *
     * @param columnsHasData
     *            Record if there is data in each column( false:null;true:has data)
     */
     @Deprecated
    static public void deleteEmptyColumns(List<Boolean> columnsHasData, Project project) throws ModelException {
    ImporterUtilities.deleteEmptyColumns(project, columnsHasData);
    }
```

Since this is technically a public method (even though it probably shouldn't have been and no one should really be using it), let's practice good hygiene and retain it with an `@Deprecated` annotation.

Reply:
re-added the original public method in `TabularImportingParserBase` with a deprecation notice (see commit  "[deprecate deleteEmptyColumns instead of removing it](https://github.com/OpenRefine/OpenRefine/pull/7272/commits/ae1ae5afc9226a24f20b4bcb4a6b3b07af7ae8bd)")